File: examples/kotlin/grammar/assignment-rule/kotlin/cott_impl/curriculum/assignment_rule/validate_access_code.kt
package cott_impl.curriculum.assignment_rule

internal fun validate_access_code(code: kotlin.String): cott_runtime.CottResult<kotlin.String, curriculum.assignment_rule.AccessCodeError> {
    val trimmed = code.trim()
    return if (trimmed.codePointCount(0, trimmed.length) >= 4) {
        cott_runtime.Ok(trimmed)
    } else {
        cott_runtime.Err(curriculum.assignment_rule.AccessCodeError.TooShort)
    }
}

File: examples/grammar/assignment-rule/python/_cott_impl/curriculum/assignment_rule/validate_access_code.py
from cott_runtime import Err, Ok, Result
from curriculum.assignment_rule_types import AccessCodeError, AccessCodeError_EmptyCode, AccessCodeError_TooShort


def validate_access_code(code: str) -> Result[str, AccessCodeError]:
    if len(code) == 0:
        return Err(error=AccessCodeError_EmptyCode())
    if len(code) < 4:
        return Err(error=AccessCodeError_TooShort())
    return Ok(value=code)
